### DISCUSSION TITLE
fill the combo_net dictionary

### DISCUSSION THREAD

**Discussion by oscarhx** [PMEAL/OpenPNM#2318] [Q&A]
Hello Professors,

Iike the demo in "examples/notebooks/networks/manipulation/coupling_continuum_regions_with_pore_networks.ipynb"
If I want to fill the combo_net dictionary from the subregion geometry and physics data, for example, how to get combo_net['throat.diameter'] or combo_net['throat.diffusive_conductance']

Thanks and Best

**Comment by oscarhx** [comment]
Hello professors,
I just detected the subdomain data can be obtained directly, it just did not show in the spy variable explorer.
Please remove this :)
Best,

**Reply by jgostick** [reply]
Can you elaborate a bit?  What do you mean by obtained directly?  Also, I don't use the spyder variable explorer, so have never noticed.

**Comment by oscarhx** [comment]
Hello Prof.&nbsp;jgostick


Thanks for your attention,
Still using the demon in the question,
in the spyder's IDE, the combo_net shows like:

there is no throat.diameter record. however, if I directly type combo_net['throat.diameter'] in the console window, It works ok:


and such dictionary data can be obtained in the output vtk data.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PMEAL/OpenPNM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月1日(星期二) 晚上9:47
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PMEAL/OpenPNM] fill the combo_net dictionary (Discussion #2318)





 
Can you elaborate a bit?  What do you mean by obtained directly?  Also, I don't use the spyder variable explorer, so have never noticed.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by oscarhx** [reply]
Image1:
![image](https://user-images.githubusercontent.com/95697681/151987522-535ba220-ade3-4724-a8b1-ab2a5d296509.png)
Image2:
![image](https://user-images.githubusercontent.com/95697681/151987680-36ebc973-3d02-4a14-9ebe-b013cc3356a5.png)

**Reply by jgostick** [reply]
OK, I see. That is expected behavior.  It's because the 'pore.diameter' is on the geometry subdomain object.  When you ask the network for it, and it doesn't find it on itself, then it searches it's associated geometry objects and interleaves them together into a single array spanning the whole domain.
